File: 209-MinimumSizeSubarraySum/209-MinimumSizeSubarraySum.py
# Last updated: 08/05/2026, 09:30:51
class Solution:
    def minSubArrayLen(self, target: int, nums: List[int]) -> int:

        left=0
        right=left
        minLength=float('inf')
        yes=False

        runSum=sum(nums[left:right+1])

        while right<len(nums):
            

            if runSum >=target:
                yes= True
                if right-left<minLength:
                    minLength=right-left
                runSum-=nums[left]
                left+=1
                
            else:

                right+=1
                if right<len(nums):
                    runSum+=nums[right]

        if yes==False:
            return 0

        return minLength+1



        

File: 209-MinimumSizeSubarraySum/209-MinimumSizeSubarraySum.cpp
// Last updated: 08/05/2026, 09:30:45
class Solution {
public:
    int minSubArrayLen(int target, vector<int>& nums) {
        int l=0;
        int r=0;
        int sum=0;
        int n= nums.size();
        int maxlen=INT_MAX;
        while(r<n){

            sum+=nums[r];
            
            while(sum>=target){
                maxlen=min(maxlen,r-l+1);
                sum-=nums[l];
                l++;
            }
            r++;
            
        }
        if(maxlen==INT_MAX){
            return 0;
        }
        return maxlen;
    }
};